crates/miden-testing/src/executor.rs

```diff
@@ -63,7 +63,7 @@ impl<H: SyncHost> CodeExecutor<H> {
     pub fn execute_program(
         mut self,
         program: Program,
-        source_manager: Arc<dyn SourceManager>,
+        source_manager: Arc<dyn SourceManager + Send + Sync + 'static>,
```

Comment:
I wonder how these extra bounds will affect the client (and specifically, WebClient). Maybe there is no effect, but would be good to double-check.

cc @igamigo.

Reply:
I don't think these new bounds would affect the client. It's essentially a stricter bound than what the web client would need, because the future that `maybe_async` generates is not `Send`, but should not make a difference. We'll still double check that everything compiles fine when we get a chance.